### DISCUSSION TITLE
Psycopg 2

### DISCUSSION THREAD

**Discussion by gitmatsika** [flet-dev/flet#3204] [Packages] [answered]
Add Psycopg 2 to flet
https://pypi.org/project/psycopg2/

**Comment by FeodorFitsner** [comment]
So, how (why?) are we supposed to access PostgreSQL on/from a mobile device? What's the use case here?

**Reply by baseplate-admin** [reply]
Design something like pgAdmin for android?

**Comment by baseplate-admin** [comment]
This is generally not a good idea to access a database from clients device. 

Databases cant handle concurrent connections like a web server can. 

So i am -1 on this

**Reply by FeodorFitsner** [reply]
That's my concern as well.

I would imaging the usage of `firebase`/`supabase` lib to access hosted/cloud instance of PGSQL as a service, but talking to SQL Server directly... 🤷‍♂️ The same could be said about SQL Server via ODBC - would be nearly impossible to make it work on a mobile.

**Reply by DavMelchi** [reply]
Even if it's not the most suitable way, it's still simple to use in some use cases than going through an API. A personal hosted database without an API. I've already done it for a personal mobile app, just to save information in a my own hosted MySQL database and write some simple sql query. With Flutter, it's possible.
 https://pub.dev/packages/postgres

**Reply by FeodorFitsner** [reply]
Do you use that flutter package to access MySQL as well?

**Reply by DavMelchi** [reply]
Nope.
https://pub.dev/packages/mysql1 for Mysql DB

**Comment by gitmatsika** [comment]
For interacting with postgres off  a mobile phone

**Comment by ndonkoHenri** [comment, marked as answer]
psycopg2 is now available for iOS and Android: https://pypi.flet.dev/psycopg2

Just add it to your dependencies and `flet build` pulls the right iOS/Android wheels automatically:

```toml
# pyproject.toml
dependencies = [
    "flet",
    "psycopg2",
]
```

Note: It's a **client** — it connects to a *remote* PostgreSQL server over the network; there's no embedded database on the phone. TLS and SCRAM-SHA-256 auth work out of the box (libpq + OpenSSL are bundled into the wheel).

Example with a real `connect()` + query round-trip against PostgreSQL 17:

```python
import psycopg2

conn = psycopg2.connect(
    host="your-db-host", port=5432, dbname="mydb", user="me", password="secret"
)
with conn.cursor() as cur:
    cur.execute("SELECT version()")
    print(cur.fetchone())
conn.close()
```

Give it a try and let know how.
